cmd/nodecmd/wiz.go

```diff
@@ -371,3 +391,88 @@ func filterHosts(hosts []*models.Host, nodes []string) ([]*models.Host, error) {
 	}
 	return filteredHosts, nil
 }
+
+func checkForSubnetValidators(
+	clusterName string,
+	subnetName string,
+) ([]string, error) {
+	ux.Logger.PrintToUser("Checking if node(s) are subnet validator(s)")
+	hosts, err := ansible.GetInventoryFromAnsibleInventoryFile(app.GetAnsibleInventoryDirPath(clusterName))
+	if err != nil {
+		return nil, err
+	}
+	if len(validators) != 0 {
+		hosts, err = filterHosts(hosts, validators)
+		if err != nil {
+			return nil, err
+		}
+	}
+	hostIDs, err := utils.MapWithError(hosts, func(h *models.Host) (string, error) {
+		_, o, err := models.HostAnsibleIDToCloudID(h.NodeID)
+		return o, err
+	})
+	if err != nil {
+		return nil, err
+	}
+	nodeIDs, err := utils.MapWithError(hostIDs, func(s string) (ids.NodeID, error) {
+		n, err := getNodeID(app.GetNodeInstanceDirPath(s))
+		return n, err
+	})
+	if err != nil {
+		return nil, err
+	}
+	network, err := getClusterNetwork(clusterName)
+	if err != nil {
+		return nil, err
+	}
+	sc, err := app.LoadSidecar(subnetName)
+	if err != nil {
+		return nil, err
+	}
+	subnetID := sc.Networks[network.Kind.String()].SubnetID
+	if subnetID == ids.Empty {
+		return nil, fmt.Errorf("expected a subnet ID different from primary network")
+	}
+	nonValidators := []string{}
+	for i, nodeID := range nodeIDs {
+		isValidator, err := subnet.IsSubnetValidator(subnetID, nodeID, network)
+		if err != nil {
+			return nil, err
+		}
+		if !isValidator {
+			nonValidators = append(nonValidators, hostIDs[i])
+		}
+	}
+	return nonValidators, nil
+}
+
+func waitForSubnetValidators(
+	clusterName string,
+	subnetName string,
+	timeout time.Duration,
+	poolTime time.Duration,
+) error {
+	ux.Logger.PrintToUser("")
+	ux.Logger.PrintToUser("Waiting for node(s) in cluster %s to become subnet validator(s) of %s...", clusterName, subnetName)
+	startTime := time.Now()
+	for {
+		nonValidators, err := checkForSubnetValidators(clusterName, subnetName)
+		if err != nil {
+			return err
+		}
+		if len(nonValidators) == 0 {
+			ux.Logger.PrintToUser("All nodes are subnet validators after %d seconds", uint32(time.Since(startTime).Seconds()))
+			return nil
+		}
+		if time.Since(startTime) > timeout {
+			ux.Logger.PrintToUser("")
+			ux.Logger.PrintToUser("Non validator Nodes")
+			for _, nonValidator := range nonValidators {
+				ux.Logger.PrintToUser("  " + nonValidator)
+			}
+			ux.Logger.PrintToUser("")
+			return fmt.Errorf("failed to verify all nodes are subnet validators after %d seconds", uint32(timeout.Seconds()))
```

Comment:
can it be replaced with `return fmt.Errorf("failed to verify all nodes as subnet validators after %f seconds", timeout.Seconds())`?